helix-core/src/main/java/org/apache/helix/manager/zk/ParticipantManager.java

```diff
@@ -514,7 +515,7 @@ public void disconnect() {
     try {
       ParticipantHistory history = getHistory();
       history.reportOffline();
-      persistHistory(history);
+      persistHistory(history, true);
```

Comment:
Why this is true? If disconnect, we are giving up the node and if instance has been dropped between zk connection disconnect and this call, we will still create that.

It would be meaning less if there was a path does not exist and update this path.

Reply:
Chatted offline, here skipOnEmptyPath=true is the expected behavior.
Updated for BaseControllerDataProvider.